File: python/models/get_unit_types_model.py
from database import get_db, rows_to_list_of_dict


def get_unit_types_model():
    try:
        db = get_db()
        rows = db.execute("SELECT * FROM unit_types").fetchall()
        unit_types = rows_to_list_of_dict(rows)
        print(f"Got {len(unit_types)} unit types")
        return unit_types
    except Exception as e:
        print("Error getting unit types from database:")
        print(e)
        return []

File: go/models/getUnitTypesModel.go
package models

import (
	"fmt"
	"hellocrud/types"
	"hellocrud/utils"

	_ "github.com/mattn/go-sqlite3"
)

func GetUnitTypesFromDb() ([]*types.UnitType, error) {
	var unitTypes []*types.UnitType
	db, err := utils.GetDatabase()
	if err != nil {
		fmt.Println("🛑 Error getting database")
		fmt.Println(err)
		return unitTypes, err
	}
	rows, err := db.Query("SELECT * FROM unit_types")
	if err != nil {
		fmt.Println("🛑 Error executing query")
		fmt.Println(err)
		return unitTypes, err
	}

	for rows.Next() {
		unitType := new(types.UnitType)
		err := rows.Scan(&unitType.UnitId, &unitType.UnitName, &unitType.UnitAttackType, &unitType.UnitDamage, &unitType.UnitHealth, &unitType.UnitCost)
		if err != nil {
			fmt.Println("🛑 Error scanning rows")
			fmt.Println(err)
		} else {
			unitTypes = append(unitTypes, unitType)
		}
	}

	rows.Close()
	db.Close()
	return unitTypes, err
}
